### PR TITLE
Fixed #760 Add SymmetricKey JNI and restructure build.gradle

### LINKED ISSUE #760: Sync iOS key derivation and encryption algorithm with iOS
iOS Spec:
Key Derivation: PBKDF2-SHA256 with 64000 Iterating Rounds
Cipher: AES/CBC/PKCS7Padding (256 bit keys)

Java Issues:
* Java (not Android) doesn't support key length > 128bit (#748)
* Java 6, 7, and Android do not support PBKDF2-SHA256 (only SHA1) support.
* PBKDF2 Java implementation is very slow specially with high iterating rounds (>1000).
* Java (But not Android) doesn't support PKCS7Padding (Only support PKCS5Padding). Based on [JDK7 documentation](http://docs.oracle.com/javase/7/docs/api/javax/crypto/Cipher.html), AES/CBC/PKCS5Padding is for 128 bit keys.

### PR DESCRIPTION
**Pull request by pasin**
- Added SymmetricKey JNI to provide native key derivation. The

- Restructure build.gradle to have build-android.gradle and build-java.gradle.

#748 #749 #760

### CODE REVIEW

File: build-java.gradle [hunk 1]
```diff
@@ -0,0 +1,251 @@
+apply plugin: 'c'
+apply plugin: 'cpp'
+apply plugin: 'java'
+apply plugin: 'maven'
+apply plugin: 'eclipse'
```
Comment (hideki) [inline build-java.gradle]: I believe eclipse plugin is not necessary.

File: build-java.gradle [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    maven { url 'http://files.couchbase.com/maven2/' }
+    mavenCentral()
+}
+
+////////////////////////////////////////////
+// For Java
+////////////////////////////////////////////
+
+dependencies {
+    testCompile 'junit:junit:4.11'
```
Comment (hideki) [inline build-java.gradle]: - version should be same with android.gradle.
- should be same order

File: build-java.gradle [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+}
+
+task generateJavadocs(type: Javadoc) {
+    source = sourceSets.main.java.srcDirs
+    List<File> pathList = new ArrayList<File>();
+    pathList.add(file('extra/doclet/doclet.jar'))
+    options.docletpath = pathList
+    options.doclet = "ExcludeDoclet"
+    options.showFromPublic()
+    exclude "org/apache/http/**", "com/couchbase/touchdb/**"
```
Comment (hideki) [inline build-java.gradle]: "com/couchbase/touchdb/**" can be removed from exclude

File: build-java.gradle [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+                buildable = false
+            }
+            binaries.all {
+                if (toolChain in Gcc) {
+                    cCompiler.args "-fPIC"
+                    cCompiler.args "-DLTC_NO_ASM"
+                }
+
+                // Workaround for Inline Assembly error in tomcrypt_macros.h
+                cCompiler.args "-DLTC_NO_ASM"
```
Comment (hideki) [inline build-java.gradle]: cCompiler.args "-DLTC_NO_ASM" defined twice

File: src/main/java/com/couchbase/lite/support/security/SymmetricKey.java [hunk 5]
```diff
@@ -216,11 +273,7 @@ private Cipher getCipher(int mode, byte[] iv) throws SymmetricKeyException {
      * @throws SymmetricKeyException
      */
     public Encryptor createEncryptor() throws SymmetricKeyException {
```
Comment (hideki) [inline src/main/java/com/couchbase/lite/support/security/SymmetricKey.java]: Is this method required?

Comment (pasin) [inline src/main/java/com/couchbase/lite/support/security/SymmetricKey.java]: I doesn't do much thing exception new Encryptor. iOS has this method which does some stuff so just tried to stay as close as the iOS but no problems, we can remove.

**Comment by pasin** [comment]
I updated the code per your review. Thanks!

**Pull request merged**